### ISSUE TITLE
feature req - query a http endpoint

### ISSUE THREAD

**Issue by tooptoop4** [trinodb/trino#704]
I am looking at ability to query results from an API (ie https://jsonplaceholder.typicode.com/posts?userId=1) but instead of just GET requests would like to send POST request and have presto query the response.

ie
https://redash.io/help/data-sources/querying/json-api
https://zappysys.com/blog/import-rest-api-json-sql-server/
https://docs.retool.com/docs/querying-via-sql
https://www.progress.com/tutorials/jdbc/query-any-rest-api-using-sql-via-jdbc-in-5-mins---getting-started-(part-1)

**Comment by tooptoop4** [comment]
maybe https://github.com/ebyhr/presto-flex solves some of this ?

**Comment by tooptoop4** [comment]
https://github.com/prestosql-rocks/presto-rest seems like a foundation! 
ie https://github.com/tooptoop4/presto-rest/blob/master/presto-rest-slack/src/main/java/rocks/prestodb/rest/slack/SlackRest.java#L54
more comments on https://groups.google.com/forum/#!topic/presto-users/t31yux2fs7E

**Comment by findepi** [comment]
See https://prestosql.io/docs/current/develop/example-http.html for (hopefully useful) example

**Comment by bitsondatadev** [comment]
I started looking into this as well a few months ago as a motivation to learn how to create my own connector. I started out with just twitter but we could easily extend this to generic rest calls.
https://github.com/prestosql/presto/compare/master...bitsondatadev:twitter-hacks

I think there are some fun examples we could use such as twitter and other rest endpoints that make for good tutorials at the very least. I'll update here if I start anything. @tooptoop4 let me know if you start on this as well!
